File: HaxballOnline/Player.py

 #  The following class describes a single player during the game. 
 #  This class is a derived class and inherits from the "CirclePhysical" base class.
 
import pygame
import math
from Ball import Ball
from CirclePhysical import CirclePhysical

class Player( CirclePhysical ):
    # Object containing player's mathods

    def __init__(self, game, px, py, number, color, nickname):
        super().__init__(game, px, py, number, 1, 15, color)
        self.nickname = nickname

    def kick(self, pos):
        # check if ball is in hitbox range
        for ball in self.game.balls:
            dist = (self.p.x - ball.p.x) ** 2 + (ball.p.y - self.p.y) ** 2
            if dist <= (self.hitbox + ball.hitbox) ** 2:
                # kick ball to given pos
                ball.v = (pos - ball.p).normalize() * (pos - ball.p).length() / 12

    def mode_ball_control(self):
        # turn down ball_control to reduce the bounce
        self.ball_control = 0.1
        self.v_max = 2 / math.pow(self.weight, 2 / 3)

    def mode_normal(self):
        # bring back normal mode
        self.ball_control = 1.0
        self.v_max = 6 / math.pow(self.weight, 2 / 3)

File: Serwer/SerwerApplication/Player.cpp
#include "stdafx.h"
#include "Ball.h"
#include "Vector2D.h"

Player::Player(GameEngine* pGame, int px, int py, int number)
	:CirclePhysical(pGame, px, py, number, 1, 15)
{
	this->team = -1;
}


Player::~Player()
{
}


void Player::kick(Vector2D mousePos)
{
	//for (Ball b : this->pGame->getBall)
	Ball* pBall = this->pGame->getBall();
	Vector2D v = mousePos - pBall->getMove();


	double dist = pow(this->getPosition().getX() - pBall->getPosition().getX(), 2);
	dist += pow(pBall->getPosition().getY() - this->getPosition().getY(), 2);

	if (dist <= pow(pBall->getHitbox() + this->getHitbox(), 2))
	{
		//kick ball to given position
		v = v.Normal(v) * v.length() / 12;
		pBall->setMove(v);
	}
}


void Player::modeBallControl()
{
	//turn down ball_control to reduce the bounce
	this->ballControl = 0.1;
	this->vMax = 2 / pow(this->getWeight(), 2 / 3);
}


void Player::modeNormal()
{
	// bring back normal mode
	this->ballControl = 1.0;
	this->vMax = 6 / pow(this->getWeight(), 2 / 3);
}


short int Player::getTeam()
{
	return team;
}


void Player::setTeam(short int team)
{
	this->team = team;
}